walrus/methods/input/process_media.py

```diff
@@ -1026,13 +1031,66 @@ def process_sensor_fusion_json(self):
             self.input.status = 'failed'
             self.input.update_log = self.log
 
+    def check_free_tier_limits(self):
```

Comment:
If open source or premium plan this should return before doing a SQL query right? Like in `check_export_billing`

On free plans, on the old method we cached the "last checked" thing, so for performance this query would only run once every x minutes or so. I think that's a good idea to keep if reasonable

Reply:
Yes in this case it should run before the queries and main processing. But we do it after the upload, that's something we still have to do because we need to count frames for videos specifically.